### PR TITLE
Feat: configure devcontainer and clean up repo

### LINKED ISSUE #335: Configure repo with devcontainer
## Background

This repo has a [`docker-compose.yml`](https://github.com/cal-itp/mobility-marketplace/blob/main/docker-compose.yml) file that makes  development and running locally fairly straightforward.

But we are trying to standardize on devcontainers with pre-commit and other tooling to streamline the dev experience and ease future onboarding.

## Tasks

* [x] `mkdir .devcontainer`
* [x] `mv docker-compose.yml .devcontainer/compose.yml`
* [x] `touch .devcontainer/devcontainer.json` - see the [Benefits `devcontainer.json`](https://github.com/cal-itp/benefits/blob/dev/.devcontainer/devcontainer.json) as an example
* [x] Rework Docker to install both jekyll (ruby) and pre-commit (python)
* [x] `touch .pre-commit-config.yaml` - see the [Benefits `.pre-commit-config.yaml`](https://github.com/cal-itp/benefits/blob/dev/.pre-commit-config.yaml) as an example
* [x] Configure a `postAttach.sh` script to setup pre-commit in the devcontainer
* [x] Update README with new steps to get started

### LINKED ISSUE #336: Clean up unused content/config
While minor, this stuff adds unnecessary build steps/time and is just clutter. As part of #335, we want a clean environment.

## Tasks

* [x] Delete directory `_articles/` (unused collection)
* [x] Delete `_layouts/articles.html` (unused collection)
* [x] Delete directory `_posts/` (unused collection)
* [x] Delete directory `admin/` (unused CMS config)
* [x] Delete directory `docs/` (HTML5 boilerplate docs)
* [x] Delete `CONDUCT.md` (need something for Cal-ITP)
* [x] Delete `humans.txt` (unmaintained)
* [x] Delete `robots.txt` (unnecessary)
* [x] In `_config.yml`:
  * [x] Delete [`authors`](https://github.com/cal-itp/mobility-marketplace/blob/main/_config.yml#L32) (unmaintained)
  * [x] Delete [`collections.articles`](https://github.com/cal-itp/mobility-marketplace/blob/main/_config.yml#L43) (unused)
  * [x] Clean up [`exclude`](https://github.com/cal-itp/mobility-marketplace/blob/main/_config.yml#L84) with the renames and deletions

### PR DESCRIPTION
**Pull request by thekaveman**
Replace basic Docker Compose setup with our standard devcontainer setup. Closes #335 

I also took this opportunity to clean up a bunch of the extra files / sample content / etc. in the repo that wasn't being used or needed anymore. Closes #336

### CODE REVIEW

**Comment by angela-tran** [comment]
Currently reviewing / trying out locally

File: .devcontainer/compose.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+services:
+  site:
+    build:
+      context: ..
+      dockerfile: .devcontainer/Dockerfile
+    command: sleep infinity
+    ports:
+      - "4000"
```
Comment (angela-tran) [inline .devcontainer/compose.yml:8]: I can't seem to connect to the site through the dynamic port. Steps I did:
 - Run `docker ps` to see the dynamic port for `mobility-marketplace_devcontainer-site-1`
 - Go to `http://localhost:<dynamic port number>`

Firefox says `The connection was reset`.

I can see the site at `http://127.0.0.1:4000`, but live-reload doesn't work.

**Review by angela-tran — COMMENTED** [review]
Nice cleanup 👏 

I ran into an issue that I also had with `compiler/compiler.la`... wondering what the behavior is in your environment.

File: .devcontainer/compose.yml (hunk 1 above)
Comment (thekaveman) [reply .devcontainer/compose.yml:8]: Yeah you are right, I'm seeing the same thing... hmmm 🤔

Comment (thekaveman) [reply .devcontainer/compose.yml:8]: Actually... LiveReload just started working 🤷 On the `http://127.0.0.1:4000` URL

Comment (angela-tran) [reply .devcontainer/compose.yml:8]: Oh interesting, it started working for me too. 🤔

File: README.md [hunk 2]
```diff
@@ -11,53 +11,44 @@ Deployed via [Netlify][netlify] [![Netlify Status](https://api.netlify.com/api/v
...
+1. Enter or select `Jekyll: Build Dev`
 
-While [Netlify][netlify] is responsible for the production site builds, you may want to replicate this process locally for development and testing.
-
-We include a `docker-compose` command that runs the official [Netlify build image][netlify-build] against the local repository contents, with our production build script(s).
-
-```bash
-docker-compose run netlify
-```
+The site is running on <http://localhost>; check the VS Code _Ports_ tab for the exact port. Auto rebuild/reload will be active
```
Comment (angela-tran) [inline README.md:42]:
```suggestion
The site is running on <http://127.0.0.1>; check the VS Code _Ports_ tab for the exact port. Auto rebuild/reload will be active
```

**Review by angela-tran — APPROVED** [review]
Suggested a tweak to the docs, but non-blocking

**Comment by thekaveman** [comment]
Merging this now so I can work on some of the recent content issues.

**Pull request merged**